File: emp_ui/lib/models/project.dart
class ProjectModel {
  final int? id;
  final String projectName;
  final String projectCode;
  final String description;
  final String startDate;
  final String endDate;
  final double budget;
  final String status;
  final String clientName;
  final String priority; 

  ProjectModel({
    this.id,
    required this.projectName,
    required this.projectCode,
    required this.description,
    required this.startDate,
    required this.endDate,
    required this.budget,
    required this.status,
    required this.clientName,
    required this.priority,
  });

  factory ProjectModel.fromJson(Map<String, dynamic> json) {
    return ProjectModel(
      id: json['projectId'],
      projectName: json['projectName'],
      projectCode: json['projectCode'],
      description: json['description'],
      startDate: json['startDate'],
      endDate: json['endDate'],
      budget: (json['budget'] as num).toDouble(),
      status: json['status'],
      clientName: json['clientName'],
      priority: json['priority'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "projectId": id,
      "projectName": projectName,
      "projectCode": projectCode,
      "description": description,
      "startDate": startDate,
      "endDate": endDate,
      "budget": budget,
      "status": status,
      "clientName": clientName,
      "priority": priority,
    };
  }
}

File: emp-api/src/main/java/com/example/entities/Project.java
package com.example.entities;

import jakarta.persistence.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "projects")
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long projectId;

    private String projectName;

    @Column(unique = true)
    private String projectCode;

    private String description;

    private LocalDate startDate;

    private LocalDate endDate;

    private BigDecimal budget;

    @Enumerated(EnumType.STRING)
    private ProjectStatus status;

    private String clientName;

    @Enumerated(EnumType.STRING)
    private Priority priority;

    // ===== ENUM =====

    public enum ProjectStatus {
        PENDING,
        RUNNING,
        COMPLETED
    }

    public enum Priority {
        HIGH,
        MEDIUM,
        LOW
    }



    // ===== Getter & Setter =====

    public Long getProjectId() {
        return projectId;
    }

    public void setProjectId(Long projectId) {
        this.projectId = projectId;
    }

    public String getProjectName() {
        return projectName;
    }

    public void setProjectName(String projectName) {
        this.projectName = projectName;
    }

    public String getProjectCode() {
        return projectCode;
    }

    public void setProjectCode(String projectCode) {
        this.projectCode = projectCode;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public LocalDate getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDate startDate) {
        this.startDate = startDate;
    }

    public LocalDate getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDate endDate) {
        this.endDate = endDate;
    }

    public BigDecimal getBudget() {
        return budget;
    }

    public void setBudget(BigDecimal budget) {
        this.budget = budget;
    }

    public ProjectStatus getStatus() {
        return status;
    }

    public void setStatus(ProjectStatus status) {
        this.status = status;
    }

    public String getClientName() {
        return clientName;
    }

    public void setClientName(String clientName) {
        this.clientName = clientName;
    }

    public Priority getPriority() {
        return priority;
    }

    public void setPriority(Priority priority) {
        this.priority = priority;
    }
}
